homeassistant/components/simplisafe/__init__.py

```diff
@@ -0,0 +1,140 @@
+"""
+Support for SimpliSafe alarm systems.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/simplisafe/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import (
+    CONF_CODE, CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_TOKEN, CONF_USERNAME)
+from homeassistant.core import callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+from homeassistant.helpers import config_validation as cv
+
+from .config_flow import configured_instances
+from .const import DATA_CLIENT, DEFAULT_SCAN_INTERVAL, DOMAIN, TOPIC_UPDATE
+
+REQUIREMENTS = ['simplisafe-python==3.1.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ACCOUNTS = 'accounts'
+
+DATA_LISTENER = 'listener'
+
+ACCOUNT_CONFIG_SCHEMA = vol.Schema({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_CODE): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL):
+        cv.time_period
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_ACCOUNTS):
+            vol.All(cv.ensure_list, [ACCOUNT_CONFIG_SCHEMA]),
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+@callback
+def _async_save_refresh_token(hass, config_entry, token):
+    config_entry.data[CONF_TOKEN] = token
```

Comment:
So this is not allowed. You need to create a new dictionary, than pass the updated dictionary to async_update_entry.